pkg/apiserver/webhooks.go

```diff
@@ -226,29 +238,130 @@ func createValidatingWebhook(ctx context.Context, webhookNs string, caCert []byt
 	return nil
 }
 
-func runWebhookServer(certStorageDir string) error {
-	certFile := filepath.Join(certStorageDir, "tls.crt")
-	keyFile := filepath.Join(certStorageDir, "tls.key")
-
-	cert, err := tls.LoadX509KeyPair(certFile, keyFile)
+// load the certificate from the secret and update when secret cert data changes e.g.
+func loadCertificate(certPath, keyPath string) (tls.Certificate, error) {
```

Comment:
Might be a good idea to extract out the cert/file handling stuff out to it's own go mod/file and add some test around it.
Also, the reloading is non trivial and could do with some test also.

Reply:
- yes i agree i will amend the previous comments first then begin drafting some test cases for this & separation to a new file.

Reply:
I also suggest to add test to test 
- if the interaction with the cert manager works as expected
- and that the webhook still works properly with the cert manager-provided certificates